components/brave_wallet_ui/ledger/ledger.ts

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { LedgerCommand, kLedgerBridgeUrl } from '../common/hardware/ledgerjs/ledger-messages'
+import { LedgerUntrustedMessagingTransport } from '../common/hardware/ledgerjs/ledger-untrusted-transport'
+
+const params = new Proxy(new URLSearchParams(window.location.search), {
+  get: (searchParams, prop) => {
+    if (typeof prop === 'string') {
+      return searchParams.get(prop)
+    }
+    return null
+  }
+})
+
+/* eslint-disable  @typescript-eslint/dot-notation */
+const targetUrl = params['targetUrl']
```

Comment:
How about this instead of the proxy?
`const targetUrl = new URLSearchParams(window.location.search).get('targetUrl')`